test/suite/helper_tiered.py

```diff
@@ -90,3 +90,82 @@ def generate_s3_prefix(test_name = ''):
 
     return prefix
 
+# Storage sources.
+tiered_storage_sources = [
+    ('dirstore', dict(is_tiered = True,
+        is_local_storage = True,
+        auth_token = get_auth_token('dir_store'),
+        bucket = get_bucket1_name('dir_store'),
+        bucket_prefix = "pfx_",
+        ss_name = 'dir_store')),
+    ('s3', dict(is_tiered = True,
+        is_local_storage = False,
+        auth_token = get_auth_token('s3_store'),
+        bucket = get_bucket1_name('s3_store'),
+        bucket_prefix = generate_s3_prefix(),
+        ss_name = 's3_store')),
+    ('non_tiered', dict(is_tiered = False)),            
+]
+
+# This mixin class provides tiered storage configuration.
+class TieredConfigMixin:
+    # Returns True if the current scenario is tiered.
+    def is_tiered_scenario(self):
+        res = False
+        if hasattr(self, 'is_tiered'):
+            res = self.is_tiered
+        
+        return res
+
+    # Setup custom connection config.
+    def conn_config(self):
+        return self.tiered_conn_config()
+
+    def tiered_conn_config(self):
+        # Handle non_tiered storage scenarios.
+        if not self.is_tiered_scenario():
+            return ''
+
+        # Setup directories structure for local tiered storage.
+        if self.is_local_storage and not os.path.exists(self.bucket):
+            os.mkdir(self.bucket)
+
+        # Build tiered storage connection string.
+        return \
+            'debug_mode=(flush_checkpoint=true),' + \
+            'tiered_storage=(auth_token=%s,' % self.auth_token + \
+            'bucket=%s,' % self.bucket + \
+            'bucket_prefix=%s,' % self.bucket_prefix + \
+            'name=%s),tiered_manager=(wait=0)' % self.ss_name
+
+    # Load the storage sources extension.
+    def conn_extensions(self, extlist):
+        return self.tiered_conn_extensions(extlist)
+
+    def tiered_conn_extensions(self, extlist):
+        # Handle non_tiered storage scenarios.
+        if not self.is_tiered_scenario():
+            return ''
+        
+        config = ''
+        # S3 store is built as an optional loadable extension, not all test environments build S3.
+        if not self.is_local_storage:
+            #config = '=(config=\"(verbose=1)\")'
+            extlist.skip_if_missing = True
+        #if self.is_local_storage:
```

Comment:
I copied the implementation of this method from `test_tiered02.py`. And it's the same across all test_tieredXX tests. That made me think that it could be done on purpose. Probably to ease adding verbose configuration? Here's the PR that introduces S3 configuration to the tests: https://github.com/wiredtiger/wiredtiger/pull/7606/files

Reply:
Hmm, interesting, @sulabhM, do you remember why?

Reply:
That is correct, to make it easier to generate debug logs when needed.

Reply:
Thanks @sulabhM!